sycl/include/sycl/ext/intel/fpga_device_selector.hpp

```diff
@@ -55,6 +59,22 @@ class fpga_emulator_selector : public platform_selector {
   fpga_emulator_selector() : platform_selector(EMULATION_PLATFORM_NAME) {}
 };
 
+class fpga_simulator_selector : public fpga_selector {
+public:
+  fpga_simulator_selector(unsigned int numDevices = 1) {
+    // Tell the runtime to start a simulator device
+    // Notes:
+    //   This replaces any hardware devices present
+    //   Currently only 1 simulator device is supported
+    auto devices = std::to_string(numDevices);
+#ifdef _WIN32
+    _putenv_s("CL_CONTEXT_MPSIM_DEVICE_INTELFPGA", devices.c_str());
+#else
+    setenv("CL_CONTEXT_MPSIM_DEVICE_INTELFPGA", devices.c_str(), 0);
```

Comment:
Can we go further and just recommend users to set `CL_CONTEXT_MPSIM_DEVICE_INTELFPGA` env var before running their app to target simulator? In this case they would not need to rewrite and recompile the code to replace `fpga_selector` with `fpga_simulator_selector` ?

Reply:
There are no plans for this current release to change this